session.go

```diff
@@ -418,6 +417,12 @@ func (s *Session) mux(remoteBegin *frames.PerformBegin) {
 				link, linkOk := s.linksByKey[linkKey{name: body.Name, role: !body.Role}]
 				s.linksMu.RUnlock()
 				if !linkOk {
+					_ = s.txFrame(&frames.PerformEnd{
```

Comment:
Hmm I think this is in violation of the protocol, i.e. we must wait for the ack before terminating the session.

Reply:
Fixed.